### ISSUE TITLE
Feature request - Show error/warning when doing pointer arithmetic with zero size types

### ISSUE THREAD

**Issue by Fusioon** [beefytech/Beef#1562]
Currently when doing pointer arithmetic with zero sized types it will crash at runtime with `EXCEPTION_INT_DIVIDE_BY_ZERO`.
It would be nice if compiler could catch these cases and show warning at least.
Code:
```
class Program
{
    struct Foo { }

    public static void Main()
    {
        void* q = (.)0x01;
        void* w = (.)0x01;

        Foo* f1 = null;
        Foo* f2 = null;
        let fr = f1 - f2;
        let r = q - w;
    }
}
```

Tested with: Nightly 05/10/2022

**Comment by bfiete** [comment]
Fixed at fa2cb7ba5657d1d10a3409838c2ead342457451f

**Comment by bfiete** [comment]
It'll result in a warning and zero value now
